### PR TITLE
Remove usage of Base internals (permute!!)

### PR DESCRIPTION
**Pull request by LilithHafner**
`permute!!` and `invpermute!!` are internal functions of Base and are likely to be [removed](https://github.com/JuliaLang/julia/issues/44869) soon. This PR makes it so that if they are gone, PooledArrays will still compile.

AFAICT, nobody uses these methods anyway, so deleting them would work too, but this approach has a more minimal impact.

### CODE REVIEW

**Comment by bkamins** [comment]
I would just remove them.

@nalimilan, @quinnj - what do you think?

**Comment by nalimilan** [comment]
If they are removed from Base I agree we can just drop them.

**Review by bkamins — APPROVED** [review]

**Comment by nalimilan** [comment]
Thanks. Let's just wait for the decision in Julia then.

**Comment by LilithHafner** [comment]
I was holding off on merging the Base PR until after this to avoid transient breakage. If you are not willing to delete these methods until the Base PR merges, then we can revert this PR to [94beed6](https://github.com/JuliaData/PooledArrays.jl/pull/87/commits/94beed602d32fa1f9af6b5f8720219c5757f3729) which should be pretty uncontroversial. FWIW, I am not aware of any packages that call the methods this PR removes.

**Comment by nalimilan** [comment]
OK, I thought that the PR was still being discussed in Julia. If all that's blocking it is PRs in packages let's merge this.

**Comment by LilithHafner** [comment]
bump

**Review by bkamins — APPROVED** [review]

**Comment by bkamins** [comment]
@LilithHafner - do you need a release. If yes - can you please bump the version of the package to 1.4.3 in this PR and then I can merge it and make a release.

**Comment by LilithHafner** [comment]
Yes, I do need a release, thank you.

**Pull request merged**
